component/prometheus/operator/podmonitors/crdmanager.go

```diff
@@ -0,0 +1,280 @@
+		DebugInfo:        map[string]*DiscoveredPodMonitor{},
+	}
+}
+
+func (c *CRDManager) Run(ctx context.Context) error {
+	c.ConfigGen = config_gen.ConfigGenerator{
+		Client: &c.Args.Client,
+	}
+
+	// Start prometheus service discovery manager
+	c.DiscoveryManager = discovery.NewManager(ctx, c.Logger, discovery.Name(c.Opts.ID))
+	go func() {
+		err := c.DiscoveryManager.Run()
+		if err != nil {
+			level.Error(c.Logger).Log("msg", "discovery manager stopped", "err", err)
+		}
+	}()
+
+	if err := c.runInformers(ctx); err != nil {
+		return err
+	}
+	level.Info(c.Logger).Log("msg", "informers  started")
+
+	// Start prometheus scrape manager.
+	flowAppendable := prometheus.NewFanout(c.Args.ForwardTo, c.Opts.ID, c.Opts.Registerer)
+	opts := &scrape.Options{}
+	c.ScrapeManager = scrape.NewManager(opts, c.Logger, flowAppendable)
+	defer c.ScrapeManager.Stop()
+	targetSetsChan := make(chan map[string][]*targetgroup.Group)
+	go func() {
+		err := c.ScrapeManager.Run(targetSetsChan)
+		level.Info(c.Logger).Log("msg", "scrape manager stopped")
+		if err != nil {
+			level.Error(c.Logger).Log("msg", "scrape manager failed", "err", err)
+		}
+	}()
+
+	// Start the target discovery loop to update the scrape manager with new targets.
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case m := <-c.DiscoveryManager.SyncCh():
+			targetSetsChan <- m
+		}
+	}
+}
+
+// runInformers starts all the informers that are required to discover PodMonitors.
+func (c *CRDManager) runInformers(ctx context.Context) error {
+	config, err := c.Args.Client.BuildRESTConfig(c.Logger)
+	if err != nil {
+		return fmt.Errorf("creating rest config: %w", err)
+	}
+
+	scheme := runtime.NewScheme()
+	for _, add := range []func(*runtime.Scheme) error{
+		v1.AddToScheme,
+	} {
+		if err := add(scheme); err != nil {
+			return fmt.Errorf("unable to register scheme: %w", err)
+		}
+	}
+
+	ls, err := c.Args.LabelSelector.BuildSelector()
+	if err != nil {
+		return fmt.Errorf("building label selector: %w", err)
+	}
+	for _, ns := range c.Args.Namespaces {
+		opts := cache.Options{
+			Scheme:    scheme,
+			Namespace: ns,
+		}
+
+		if ls != labels.Nothing() {
+			opts.DefaultSelector.Label = ls
+		}
+		cache, err := cache.New(config, opts)
+		if err != nil {
+			return err
+		}
+
+		informers := cache
+
+		go func() {
+			err := informers.Start(ctx)
+			if err != nil && ctx.Err() != nil {
+				level.Error(c.Logger).Log("msg", "failed to start informers", "err", err)
+			}
+		}()
+		if !informers.WaitForCacheSync(ctx) {
+			return fmt.Errorf("informer caches failed to sync")
+		}
+		if err := c.configureInformers(ctx, informers); err != nil {
+			return fmt.Errorf("failed to configure informers: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// configureInformers configures the informers for the CRDManager to watch for PodMonitors changes.
+func (c *CRDManager) configureInformers(ctx context.Context, informers cache.Informers) error {
+	objects := []client.Object{
+		&v1.PodMonitor{},
+	}
```

Comment:
Do we intend to add more objects to this slice or was this left by mistake after a refactor? 

If we don't intend to ever add more objects, it might be a good opportunity to clean this up and remove the loop.

(But IIUC, we might add more things to here eventually like Secrets or ConfigMaps).

Reply:
I am not sure about the intent here, I will let @captncraig chime in on this one